infra/storage/file_transfer_managers/s3Mgr.class.php

```diff
@@ -334,3 +374,41 @@ public function registerStreamWrapper()
 		$this->s3->registerStreamWrapper();
 	}
 }
+
+class RefreshableRole extends AbstractRefreshableCredentials
+{
+	public function refresh()
+	{
+		$credentialsCacheDir = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 's3_creds_cache';
+		
+		$credentials = new Credentials('', '');
+		$ipRefresh = new RefreshableInstanceProfileCredentials(new Credentials('', '', '', 1));
+		$ipCache = new DoctrineCacheAdapter(new FilesystemCache("$credentialsCacheDir/instanceProfileCache"));
+		
+		$ipCreds = new CacheableCredentials($ipRefresh, $ipCache, 'refresh_role_creds_key');
+		$sts = StsClient::factory(array(
+			'credentials' => $ipCreds,
+		));
+		
+		$call = $sts->assumeRole(array(
+			'RoleArn' => getenv("S3_ARN_ROLE"),
+			'RoleSessionName' => 'kaltura_s3_access',
```

Comment:
const 
if this is just session name and its valid for one hour - lets add some value to the name like hour in the day {00-24} it can be very helpful for later debug

Reply:
Done.